manifests/tunnel.pp

```diff
@@ -11,45 +11,58 @@
 # detailed description.
 
 define wireguard::tunnel (
+  Enum['present','absent'] $ensure = 'present',
   String  $private_key,
   Integer $listen_port,
-  Array[Hash] $peers,
+  String  $virtual_ip,
```

Comment:
As described above, I'd like to keep IP configuration itself away from this puppet module. I do use [`systemd::networkd::network`](https://github.com/flokli/puppet-systemd/blob/master/manifests/networkd/network.pp) to configure them, btw.